### DISCUSSION TITLE
Flow Test Summary only allow 3 test per Request

### DISCUSSION THREAD

**Discussion by DianaMallida** [postmanlabs/postman-flows#248] [Defect]
When I'm using flow test on a request with 4 or more tests, they dont appear on the summary, it only shows the first 3

**Comment by saswatds** [comment]
Hey @DianaMallida, thanks for reporting this. Let me take a look at this.

**Reply by muhfaris** [reply]
I have the same issue, any update on this issue?

**Comment by koskeller** [comment]
Any news? We have the same issue.

**Reply by rjammart** [reply]
We have also the same issue.

**Reply by dev-kanishk** [reply]
Hey, I was not able to reproduce the issue, can anyone tell me which version of Postman App you are using or it is Postman web.
Also if you could share the screen of flows. It will be really helpful.

**Reply by koskeller** [reply]
@dev-kanishk I have this issue with Postman App for Mac:
```
Version 9.15.11
UI Version: 9.15.11-ui-220404-0850
Desktop Platform Version: 9.15.2 (9.15.2)
```
Just checked Postman Web, everything works well, all tests are displayed. 

Here are two screenshots of the same flow. One from web, another one from app.
Web - https://postimg.cc/w73665t7
App - https://postimg.cc/f3QZj689

**Reply by rjammart** [reply]
This is my setup:
```
Postman for Linux
Version 9.15.11
UI Version 9.15.11-ui-220404-0850
Desktop Platform Version 9.15.2
Architecture x64
OS Platform linux 5.4.0-107-generic
```
To reproduce, I think it is enough to have more than 3 tests in a request, call the request from a flow and plug a "Test summary" component. You should see that only 3 tests appears on the "logs".

**Comment by grrigore** [comment]
Any updates on this? I too get only three results and I have some extra tests defined for each request.

**Reply by ShortyLago** [reply]
I have the same issue. Any update so far pls?

**Comment by ShortyLago** [comment]
I also found out that if there is a for loop in test it won't be executed at all and also any test after this one.
For instance I have 4 tests "pm.test();" blocks where the third one has for loop. This is the result what I can see in Test Summary of Flows:
1. Passed
2. Passed

As soon as it hits pm.test(); block with for loop in it, it won't execute any test block further away.

**Comment by JuanScerri** [comment]
Hello. I'm experiencing a similar issue with regards to the test summary not running all the tests. You can find a demonstration below.

https://user-images.githubusercontent.com/111490976/185556536-d7ef2df7-0fed-4ab7-9962-7e673727b868.mp4

From further research it seems to be completely halting the test script execution after the first test because additional logging (with console.log) was not being logged to the console after the first test.

**Comment by hobk1ard** [comment]
Flow Test Summary and Terminal both only seem to record the first three tests; however all tests are recorded when running the same Post request manually through collections or Test Runner.

**Comment by cstef123** [comment]
I am running into the same issue, any update? Flows are great, but this is a huge issue, there is no point to use if you cannot see results. 

Here is my issue: 

Flow Test Summary only record the first four tests in the Postman Desktop Application (Version 10.0.39 for windows), but there are 10 tests total. 

But, when running the same flow in the web application all 10 test cases show in the Test Summary.

**Comment by CharlieSnowCode** [comment]
I've noticed a related issue.  We have 2 tests at the Collection level (status code and response time) and a set of validations in a test in a  Request.  When I run a Flow for that Request, I only see the Collection level results in the Terminal and Test Summary Blocks.  If I comment out the tests in the Collection, I can then see the results of the test from the Request.

**Comment by NikhilM42** [comment]
I am seeing this defect also a year later since it was reported, not a show stopper as I can bundle my tests, but I would rather not.
Scenario is I have a request block that is triggered by a for loop block. The request block has 4 tests and I am only seeing a maximum of 3 tests come through when the block fails one of the tests. This issue is regardless of whether I am on the web or on the desktop app. Let me know if you need more information to debug or if you need me to build out a mini flow that recreates the issue. Image below for any version information.

![image](https://user-images.githubusercontent.com/11237528/218600321-dfb09044-090e-49aa-97c2-596802434c10.png)

**Reply by viko42** [reply]
You should consider creating a new ticket here: https://github.com/postmanlabs/postman-app-support/labels/Flows

**Reply by NikhilM42** [reply]
Done, ticket can be found here https://github.com/postmanlabs/postman-app-support/issues/11734
